app/controllers/application_controller.rb

```diff
@@ -17,4 +17,19 @@ def configure_permitted_parameters
   def after_sign_out_path_for(resource_or_scope)
     new_user_session_path
   end
+
+  def default_url_options
+    {locale: I18n.locale}
+  end
+  
+  def set_locale
+    I18n.locale = take_locale || I18n.default_locale
+  end
+  
+  def take_locale
+    parsed_locale = params[:locale]
+    I18n.available_locales.map(&:to_s).include?(parsed_locale) ? parsed_locale : nil
+  end
```

Comment:
```suggestion
  def set_locale
    I18n.locale = I18n.available_locales.include?(params[:locale].to_sym) && params[:locale] || I18n.default_locale
  end
```

Reply:
Actually that gives me that error :  
`undefined method `to_sym' for nil:NilClass`
 if the url is 0.0.0.0:3000 it's is working if we specify language 0.0.0.0:3000\en
That way is working better:
` def set_locale
  I18n.locale = I18n.available_locales.map(&:to_sym).include?(params[:locale]) && params[:locale] || I18n.default_locale
  end
  `